### PR TITLE
[RAPTOR-10746] fix java and genai tests

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
1. rename constant/argument `java` - > `java_codegen` to match the name of the env folder, for easier usage.
2. fix genai tests. They have always been skipped, e.g
Prev run: 401 collected/ 401 skipped
 https://jenkins2.devinfra.drdev.io/job/MLOps/job/datarobot-user-models/job/PR-1005/1/artifact/test_functional_python311_genai.log.html

Run in the current PR: 401 collected/ 2 passed / 399 skipped
https://jenkins2.devinfra.drdev.io/job/MLOps/job/datarobot-user-models/job/PR-1010/4/artifact/test_functional_python311_genai.log.html

### CODE REVIEW

**Review by scottp-ml — APPROVED** [review]

**Comment by yakov-g** [comment]
jenkins run test_functional_r_environment

File: tests/conftest.py [hunk 1]
```diff
@@ -328,7 +328,7 @@ def framework_env(pytestconfig):
         TESTS_DATA_PATH, "10k_diabetes_sample.csv"
     ),
     (None, MULTICLASS_LABEL_SPACES): os.path.join(TESTS_DATA_PATH, "iris_with_spaces_full.csv"),
-    (SKLEARN, TEXT_GENERATION): os.path.join(TESTS_DATA_PATH, "prompts.csv"),
+    (PYTHON_TEXT_GENERATION, TEXT_GENERATION): os.path.join(TESTS_DATA_PATH, "prompts.csv"),
```
Comment (rvorobii) [inline tests/conftest.py:331]: why is this renamed?

Comment (yakov-g) [reply tests/conftest.py:331]: Because we define dataset for the (framework, problem) pair.
And it was just defined wrong.

**Review by rvorobii — APPROVED** [review]
